<h1>Activity 1 - Palmer Penguins Data Analytics Project</h1> 
<hr>
<h3> Analyst: Kim David Penaredonda</h3>

In [1]:
#import libraries 
import numpy as np 
import pandas as pd 
from scipy.stats import f_oneway
import warnings
warnings.filterwarnings('ignore')

<h1>  Converting CSV into a Data Frame </h1>

In [2]:
penguins = pd.read_csv('penguins.csv')
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h1>Describing Data</h1>

In [3]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

In [4]:
penguins.describe().T#transpose 

count         mean         std     min       25%      50%  \
rowid              344.0   172.500000   99.448479     1.0    86.750   172.50   
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                       75%     max  
rowid               258.25   344.0  
bill_length_mm       48.50    59.6  
bill_depth_mm        18.70    21.5  
flipper_length_mm   213.00   231.0  
body_mass_g        4750.00  6300.0  
year               2009.00  2009.0

<h1> Checkcing Null Values</h1>

In [5]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>Checking Value Counts</h1>

In [6]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [7]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Filtering Data</h1>

In [8]:
adelie_DF = penguins[penguins['species'] == "Adelie"]
adelie_DF

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

In [9]:
gentoo_DF = penguins[penguins['species'] == "Gentoo"]
gentoo_DF

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

In [10]:
chinstrap_DF = penguins[penguins['species'] == "Chinstrap"]
chinstrap_DF

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

<h1>Questions</h1>
<hr>
<p>How many of these Male Chinstrap Species weight is above 400grams?</p>

In [11]:
heavy_chinstrap = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

In [12]:
heavy_chinstrap

rowid    species island  bill_length_mm  bill_depth_mm  \
284    285  Chinstrap  Dream            46.0           18.9   
289    290  Chinstrap  Dream            52.0           18.1   
291    292  Chinstrap  Dream            50.5           19.6   
295    296  Chinstrap  Dream            49.2           18.2   
301    302  Chinstrap  Dream            52.0           19.0   
305    306  Chinstrap  Dream            52.8           20.0   
307    308  Chinstrap  Dream            54.2           20.8   
309    310  Chinstrap  Dream            51.0           18.8   
313    314  Chinstrap  Dream            52.0           20.7   
315    316  Chinstrap  Dream            53.5           19.9   
321    322  Chinstrap  Dream            50.8           18.5   
323    324  Chinstrap  Dream            49.0           19.6   
329    330  Chinstrap  Dream            50.7           19.7   
333    334  Chinstrap  Dream            49.3           19.9   
339    340  Chinstrap  Dream            55.8           19.8   
342    343  Chinstrap  Dream            50.8           19.0   

     flipper_length_mm  body_mass_g     sex  year  
284              195.0       4150.0  female  2007  
289              201.0       4050.0    male  2007  
291              201.0       4050.0    male  2007  
295              195.0       4400.0    male  2007  
301              197.0       4150.0    male  2007  
305              205.0       4550.0    male  2008  
307              201.0       4300.0    male  2008  
309              203.0       4100.0    male  2008  
313              210.0       4800.0    male  2008  
315              205.0       4500.0    male  2008  
321              201.0       4450.0    male  2009  
323              212.0       4300.0    male  2009  
329              203.0       4050.0    male  2009  
333              203.0       4050.0    male  2009  
339              207.0       4000.0    male  2009  
342              210.0       4100.0    male  2009

<h1>Filling Up Null Values</h1>

In [13]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [14]:
average_bill_len = penguins['bill_length_mm'].mean()
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(average_bill_len)
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [15]:
average_bill_depth = penguins['bill_depth_mm'].mean()
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(average_bill_depth)
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [16]:
average_flipper_ave = penguins['flipper_length_mm'].mean()
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(average_flipper_ave)
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [17]:
average_body_mass = penguins['body_mass_g'].mean()

In [18]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(average_body_mass)
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [19]:
frequency_sex = penguins['sex'].mode()
penguins['sex'] = penguins['sex'].fillna(frequency_sex[0])
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [20]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     344 non-null    float64
 4   bill_depth_mm      344 non-null    float64
 5   flipper_length_mm  344 non-null    float64
 6   body_mass_g        344 non-null    float64
 7   sex                344 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB


In [21]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category') 
penguins['sex'] = penguins['sex'].astype('category')
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(2)
memory usage: 17.6 KB


<h1> Exploratory Data Analysis</h1>

<h1>1. What is the highest body mass of all penguin Species?</h1>

In [22]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  
0           181.000000  3750.000000    male  2007  
1           186.000000  3800.000000  female  2007  
2           195.000000  3250.000000  female  2007  
3           200.915205  4201.754386    male  2007  
4           193.000000  3450.000000  female  2007  
..                 ...          ...     ...   ...  
339         207.000000  4000.000000    male  2009  
340         202.000000  3400.000000  female  2009  
341         193.000000  3775.000000    male  2009  
342         210.000000  4100.000000    male  2009  
343         198.000000  3775.000000  female  2009  

[344 rows x 9 columns]

In [23]:
penguins["species"].unique()

['Adelie', 'Gentoo', 'Chinstrap']
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [24]:
np.array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [25]:
avg_body_mass = penguins.groupby("species")["body_mass_g"].mean().round(2)

In [26]:
avg_body_mass

species
Adelie       3703.96
Chinstrap    3733.09
Gentoo       5068.97
Name: body_mass_g, dtype: float64

<h1>INSIGHT</h1>
<hr>
<p> Based on its outcome we can see that the Gentoo penguins have the highest body mass compared to other specie of penguins.</p>

<h1>2. What specie have the longest flipper length?</h1>

In [27]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  
0           181.000000  3750.000000    male  2007  
1           186.000000  3800.000000  female  2007  
2           195.000000  3250.000000  female  2007  
3           200.915205  4201.754386    male  2007  
4           193.000000  3450.000000  female  2007  
..                 ...          ...     ...   ...  
339         207.000000  4000.000000    male  2009  
340         202.000000  3400.000000  female  2009  
341         193.000000  3775.000000    male  2009  
342         210.000000  4100.000000    male  2009  
343         198.000000  3775.000000  female  2009  

[344 rows x 9 columns]

In [28]:
penguins[["flipper_length_mm", "species"]].value_counts()

flipper_length_mm  species  
190.0              Adelie       19
215.0              Gentoo       12
195.0              Adelie       11
193.0              Adelie       10
210.0              Gentoo       10
                                ..
205.0              Adelie        1
206.0              Chinstrap     1
207.0              Chinstrap     1
                   Gentoo        1
231.0              Gentoo        1
Name: count, Length: 84, dtype: int64

In [29]:
flipper_averages = (
    'Adelie:', round(float(penguins[penguins["species"] == "Adelie"]["flipper_length_mm"].mean()), 2),
    'Chinstrap:', round(float(penguins[penguins["species"] == "Chinstrap"]["flipper_length_mm"].mean()), 2),
    'Gentoo:', round(float(penguins[penguins["species"] == "Gentoo"]["flipper_length_mm"].mean()), 2)
)

In [30]:
flipper_averages

('Adelie:', 190.03, 'Chinstrap:', 195.82, 'Gentoo:', 217.06)

<h3>INSIGHT</h3>
<hr>
<p>The species of penguin that has the longest flipper length is Gentoo compared to other species of penguin</p>

<h1>3. In what year has the heaviest penguin?</h1>

In [31]:
max_bmg = penguins['body_mass_g'].max()
print(max_bmg)
penguins[(penguins['year']) & (penguins['body_mass_g'] >= max_bmg)]

6300.0


rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169    170  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  
169       6300.0  male  2007

<h1>INSIGHT</h1>
<hr>
<p>The graph shows the heaviest penguin ever recorded was in 2007.</p>

<h1>4.What is the average bill length for each specie?</h1>

In [32]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  
0           181.000000  3750.000000    male  2007  
1           186.000000  3800.000000  female  2007  
2           195.000000  3250.000000  female  2007  
3           200.915205  4201.754386    male  2007  
4           193.000000  3450.000000  female  2007  
..                 ...          ...     ...   ...  
339         207.000000  4000.000000    male  2009  
340         202.000000  3400.000000  female  2009  
341         193.000000  3775.000000    male  2009  
342         210.000000  4100.000000    male  2009  
343         198.000000  3775.000000  female  2009  

[344 rows x 9 columns]

In [33]:
penguins[["bill_length_mm", "species"]].value_counts()

bill_length_mm  species  
41.1            Adelie       7
37.8            Adelie       5
39.6            Adelie       5
39.7            Adelie       4
38.1            Adelie       4
                            ..
44.0            Gentoo       1
44.4            Gentoo       1
45.0            Gentoo       1
45.4            Chinstrap    1
59.6            Gentoo       1
Name: count, Length: 210, dtype: int64

<h1>INSIGHT</h1>
<hr
<p>This shows that the Chinstrap Specie has the longest bill length among species.</p>

In [34]:
# adelie_data

<h1>5. What sex have the shortest bill length?</h1>

In [35]:
 penguins['bill_length_mm'].min()

np.float64(32.1)

In [36]:
penguins[penguins['bill_length_mm'] == 32.1]

rowid species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
142    143  Adelie  Dream            32.1           15.5              188.0   

     body_mass_g     sex  year  
142       3050.0  female  2009

<h1>Insight</h1>
<hr>
<p>Based on this, Adelie females have the shortest bill length, with the minimum recorded being 32.1 mm</p>

<h1>6.Which island has the most specie of Gentoo penguins?</h1>

In [37]:
penguins[penguins['species'] == 'Gentoo']['island'].value_counts()

island
Biscoe       124
Dream          0
Torgersen      0
Name: count, dtype: int64

<h1>INSIGHT</h1>
<hr>
<p>This graph shows that gentoo penguins mostly live in biscoe islands.
</p>

<h1>7. What is the lightest body mass of female adelie in the year 2008?</h1>

In [38]:
female_adelie_2008 = penguins[(penguins["species"] == "Adelie") & (penguins["sex"] == "female") & (penguins["year"] == 2008)]
lightest_mass = female_adelie_2008["body_mass_g"].min()

In [39]:
female_adelie_2008

rowid species     island  bill_length_mm  bill_depth_mm  \
50     51  Adelie     Biscoe            39.6           17.7   
52     53  Adelie     Biscoe            35.0           17.9   
54     55  Adelie     Biscoe            34.5           18.1   
56     57  Adelie     Biscoe            39.0           17.5   
58     59  Adelie     Biscoe            36.5           16.6   
60     61  Adelie     Biscoe            35.7           16.9   
62     63  Adelie     Biscoe            37.6           17.0   
64     65  Adelie     Biscoe            36.4           17.1   
66     67  Adelie     Biscoe            35.5           16.2   
68     69  Adelie  Torgersen            35.9           16.6   
70     71  Adelie  Torgersen            33.5           19.0   
72     73  Adelie  Torgersen            39.6           17.2   
74     75  Adelie  Torgersen            35.5           17.5   
76     77  Adelie  Torgersen            40.9           16.8   
78     79  Adelie  Torgersen            36.2           16.1   
80     81  Adelie  Torgersen            34.6           17.2   
82     83  Adelie  Torgersen            36.7           18.8   
84     85  Adelie      Dream            37.3           17.8   
87     88  Adelie      Dream            36.9           18.6   
89     90  Adelie      Dream            38.9           18.8   
90     91  Adelie      Dream            35.7           18.0   
92     93  Adelie      Dream            34.0           17.1   
94     95  Adelie      Dream            36.2           17.3   
96     97  Adelie      Dream            38.1           18.6   
98     99  Adelie      Dream            33.1           16.1   

    flipper_length_mm  body_mass_g     sex  year  
50              186.0       3500.0  female  2008  
52              190.0       3450.0  female  2008  
54              187.0       2900.0  female  2008  
56              186.0       3550.0  female  2008  
58              181.0       2850.0  female  2008  
60              185.0       3150.0  female  2008  
62              185.0       3600.0  female  2008  
64              184.0       2850.0  female  2008  
66              195.0       3350.0  female  2008  
68              190.0       3050.0  female  2008  
70              190.0       3600.0  female  2008  
72              196.0       3550.0  female  2008  
74              190.0       3700.0  female  2008  
76              191.0       3700.0  female  2008  
78              187.0       3550.0  female  2008  
80              189.0       3200.0  female  2008  
82              187.0       3800.0  female  2008  
84              191.0       3350.0  female  2008  
87              189.0       3500.0  female  2008  
89              190.0       3600.0  female  2008  
90              202.0       3550.0  female  2008  
92              185.0       3400.0  female  2008  
94              187.0       3300.0  female  2008  
96              190.0       3700.0  female  2008  
98              178.0       2900.0  female  2008

In [40]:
lightest_mass

np.float64(2850.0)

<h1>INSIGHT</h1>
<hr>
<p>The result shows that the lightest mass of adelie specie in year 2008 is 2.8kg body mass.</p>

<h1>Data Aggregation</h1>


<h1>8) Which sex of Gentoo penguins has the highest average body mass?</h1>


In [41]:
penguins.groupby('sex')['body_mass_g'].max()

sex
female    5200.0
male      6300.0
Name: body_mass_g, dtype: float64

<h1>Insight</h1>
<Hr>
<P>This shows that the male Gentoo penguins has the highest average body mass</P>

<h1>9) Which species has the maximum flipper length</h1>

In [42]:
penguins.groupby('species')['flipper_length_mm'].max()

species
Adelie       210.0
Chinstrap    212.0
Gentoo       231.0
Name: flipper_length_mm, dtype: float64


penguins['flipper_length_mm'].max()

<h1><span style="color: blue;">Insight</span></h1>
<hr>
<p>the result shows that the <b>Gentoo Species</b> has the <b>maximum flipper length</b>.</p>

<h1>10)  What is maximum flipper length per group</h1>

In [51]:
penguins.groupby(['species', 'island'])['flipper_length_mm'].agg([ 'max'])

max
species   island          
Adelie    Biscoe     203.0
          Dream      208.0
          Torgersen  210.0
Chinstrap Biscoe       NaN
          Dream      212.0
          Torgersen    NaN
Gentoo    Biscoe     231.0
          Dream        NaN
          Torgersen    NaN

<h1><span style="color: blue;">Insight</span></h1>
<hr>
<p>the maximum flipper length per group: for the group of <b><i>adelie species</i></b> the maximum flipper length is <b>210</b>. for the group of <b>chinstrap</b> the maximum flipper length is <b>212</b>. lastly the maximum length for the group of <b>gentoo species</b> is <b>231</b>.</p>

<h1>11) Does the length of flippers can affect to the length of their bill?</h1>

In [54]:
corr_flipper_bill = penguins[['flipper_length_mm','bill_length_mm']].corr()
corr_flipper_bill

flipper_length_mm  bill_length_mm
flipper_length_mm           1.000000        0.656181
bill_length_mm              0.656181        1.000000

<h1><span style="color: blue;">Insight</span></h1>
<hr>
<p>the result shows that the length of their flipper could actually affect the length of their bill.</p>

<h1>12) Which year had the smallest average bill depth??</h1>

In [55]:
bill_year = penguins.groupby('year')['bill_depth_mm'].mean()
bill_year

year
2007    17.425011
2008    16.914035
2009    17.125426
Name: bill_depth_mm, dtype: float64

<h1><span style="color: blue;">Insight</span></h1>
<hr>
<p>The result shows that in year <b><i>2008</i></b> they recorded the smallest bill depth of the Palmer Penguins. </p>

<h1>13)Is there a difference in the bill length between the three penguin species?</h1>

In [52]:
bill_length, p_value = f_oneway(penguins[penguins['species'] == 'Adelie']['bill_length_mm'], 
                           penguins[penguins['species'] == 'Chinstrap']['bill_length_mm'], 
                           penguins[penguins['species'] == 'Gentoo']['bill_length_mm'])

In [53]:
p_value

np.float64(7.233266188149441e-91)

<h1><span style="color: blue;">Insight</span></h1>
<hr>
<p>The result shows that thedifference in the bill length between the three pernguin species is <b><i>7.23</i></b></p>